src/python/pants/engine/internals/parser.py

```diff
@@ -199,10 +199,15 @@ def get_env(self, name: str, *args, **kwargs) -> Any:
         """
     )
     def set_defaults(
-        self, *args: SetDefaultsT, ignore_unknown_fields: bool = False, **kwargs
+        self,
+        *args: SetDefaultsT,
+        ignore_unknown_fields: bool = False,
+        ignore_unknown_targets: bool = False,
     ) -> None:
         self.defaults.set_defaults(
-            *args, ignore_unknown_fields=self.is_bootstrap or ignore_unknown_fields, **kwargs
+            *args,
+            ignore_unknown_fields=self.is_bootstrap or ignore_unknown_fields,
+            ignore_unknown_targets=self.is_bootstrap or ignore_unknown_targets,
```

Comment:
This looks like the key line for this fix right? If it's bootstrap, then it ignores unknown targets.

Reply:
It's where the decision is made yes, I'd say the key is the implementation in `defaults.py:178-179`.